tutorials/pyroot/pyroot002_TTreeAsMatrix.py

```diff
@@ -0,0 +1,70 @@
+## \file
+## \ingroup tutorial_pyroot
+## \notebook -nodraw
+## This tutorial shows how a TTree can be quickly converted to a numpy array.
+##
+## \macro_code
+##
+## \date April 2018
+## \author Stefan Wunsch
+
+import ROOT
+from sys import exit
+
+
+def make_example():
+    root_file = ROOT.TFile("pyroot002_example.root", "RECREATE")
+    tree = ROOT.TTree("tree", "tutorial")
+    x = np.empty((1), dtype="float32")
+    y = np.empty((1), dtype="float32")
+    tree.Branch("x", x, "x/F")
+    tree.Branch("y", y, "y/F")
+
+    for i in range(4):
+        x[0] = i
+        y[0] = -i
+        tree.Fill()
+    root_file.Write()
+
+    return root_file, tree
+
+
+def main():
+    # The conversion of the TTree to a numpy array is implemented with multi-
+    # thread support.
+    ROOT.ROOT.EnableImplicitMT()
+
+    # Create a ROOT file with a tree and the branches "x" and "y"
+    _, tree = make_example()
+
+    # Print content of the tree by looping explicitely
+    print("Tree content:\n{}".format(
+        np.asarray([[tree.x, tree.y] for event in tree])))
+
+    # Read-out full tree as numpy array
+    array = tree.AsMatrix()
+    print("Tree converted to a numpy array:\n{}".format(array))
+
+    # Get numpy array and according labels of the columns
+    labels, array = tree.AsMatrix(return_labels=True)
```

Comment:
Did we get inspiration from somewhere for this signature?

Reply:
The feature can be still discussed. Let's more the discussion to #1872.

Reply:
for me it's in principle fine as long as it does not cost too much in terms of implementation :) @etejedor what do you think?

Reply:
I believe it could be useful to know the column names, and it is very cheap in terms of logic (we already have constructed a list with those names).

Reply:
Done.